test/extended/networking/kubevirt/console.go

```diff
@@ -189,69 +192,83 @@ func LoginToFedora(virtctlPath, vmiNamespace, vmiName, user, password string) er
 
 // LoginToFedora performs a console login to a Fedora base VM
 func LoginToFedoraWithHostname(virtctlPath, vmiNamespace, vmiName, user, password, hostname string) error {
-	expecter, _, err := newExpecter(virtctlPath, vmiNamespace, vmiName, consoleConnectionTimeout, expect.Verbose(true), expect.VerboseWriter(GinkgoWriter))
-	if err != nil {
-		return err
-	}
-	defer expecter.Close()
+	err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			return strings.Contains(err.Error(), "expect: Process not running")
+		},
+		func() error {
+			expecter, _, err := newExpecter(virtctlPath, vmiNamespace, vmiName, consoleConnectionTimeout,
+				expect.Verbose(true),
+				expect.VerboseWriter(GinkgoWriter),
+				expect.DebugCheck(log.New(GinkgoWriter, "expect: virtctl console check", 0 /* no flags */)),
```

Comment:
what does this do @qinqon ? Just log that we have to re-try establishing the connection ?

Reply:
It runs the check that of "virtctl command" running and prints it and return back the result, so if it fails we will se the stdout of the virtcl command itself.